cpp/core/operators/shuffle/SplitterBase.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "operators/shuffle/type.h"
+
+namespace gluten {
+
+class SplitterBase {
+ public:
+  /**
+   * Spill for fixed size of partition data
+   */
+  virtual arrow::Status SpillFixedSize(int64_t size, int64_t* actual) = 0;
```

Comment:
If it is a Splitter using RSS, generally there will be no shuffle spill, so this part can be left to subclasses to define.

Reply:
It is used in https://github.com/oap-project/gluten/blob/main/cpp/core/jni/JniWrapper.cc#L805, so I should put it in current location, we cannot access it if we left it to subclasse such as `VeloxSplitter`. You can add a new class on your demand.

Reply:
PR link: https://github.com/oap-project/gluten/pull/1102